examples/pennylane_run_doubly_stochastic.py

```diff
@@ -0,0 +1,371 @@
+    cost_SGD100.append(qnode_analytic(params_SGD100))
+    params_SGD100 = opt.step(qnode_stochastic, params_SGD100)
+
+
+##############################################################################
+# Note that in all three cases above, we keep track of the cost function
+# by evaluating the *analytic* device with the current parameter values.
+# If evaluating on hardware where analytic simulation is intractable, but
+# the optimization convergence needs to be tracked, alternative approaches
+# could include:
+#
+# * Evaluating the cost function with a larger number of samples at specified
+#   intervals,
+#
+# * Keeping track of the *moving average* of the low-shot cost evaluations.
+#
+# We can now plot the cost against optimization step for the three cases above.
+
+from matplotlib import pyplot as plt
+
+plt.style.use("seaborn")
+plt.plot(cost_GD[:100], label="Vanilla gradient descent")
+plt.plot(cost_SGD100[:100], "--", label="QSGD (100 shots)")
+plt.plot(cost_SGD1[:100], "-.", label="QSGD (1 shot)")
+
+# analytic ground state
+min_energy = min(np.linalg.eigvalsh(H))
+plt.hlines(min_energy, 0, 100, linestyles=":", label="Ground-state energy")
+
+plt.ylabel("Cost function value")
+plt.xlabel("Optimization steps")
+plt.legend()
+plt.show()
+
+##############################################################################
+# Using the trained parameters from each optimization strategy, we can
+# evaluate the analytic quantum device:
+
+print("Vanilla gradient descent min energy = ", qnode_analytic(params_GD))
+print("Stochastic gradient descent (shots=100) min energy = ", qnode_analytic(params_SGD100))
+print("Stochastic gradient descent (shots=1) min energy = ", qnode_analytic(params_SGD1))
+
+
+##############################################################################
+# Amazingly, we see that even the ``shots=1`` optimization converged
+# to a reasonably close approximation of the ground-state energy!
+
+##############################################################################
+# Doubly stochastic gradient descent for VQE
+# ------------------------------------------
+#
+# As noted in `Sweke et al. (2019) <https://arxiv.org/abs/1910.01155>`__,
+# variational quantum algorithms often include terms consisting of linear combinations
+# of expectation values. This is true of the parameter-shift rule (where the
+# gradient of each parameter is determined by shifting the parameter by macroscopic
+# amounts and taking the difference), as well as VQE, where the Hamiltonian
+# is usually decomposed into a sum of Pauli expectation values.
+#
+# Consider the Hamiltonian from the previous section. As this Hamiltonian is a
+# Hermitian observable, we can always express it as a sum of Pauli matrices using
+# the relation
+#
+# .. math::
+#
+#     H = \sum_{i,j=0,1,2,3} a_{i,j} (\sigma_i\otimes \sigma_j),
+#
+# where
+#
+# .. math::
+#
+#     a_{i,j} = \frac{1}{4}\text{tr}[(\sigma_i\otimes \sigma_j )H], ~~ \sigma = \{I, X, Y, Z\}.

```

Comment:
Can you double check the fraction is correct here? It should be compatible with whatever prefactor makes sense for the Pauli gates as implemented in pennylane

Reply:
Yep, this is correct! Did the multiplication out.